### PR TITLE
:sparkles: add basic aws calculations

### PR DESCRIPTION
**Pull request by bcesur**

### CODE REVIEW

File: pkg/agent/targets/aws/target.go [hunk 1]
```diff
@@ -72,9 +89,79 @@ func (Target) Register(a any, r *prometheus.Registry) error {
...
+	for k, v := range cpuUtilizations {
+		calculationsGauge.WithLabelValues(
+			k, v.AvgWatts, v.ComputedWattHours)
+
+	}
+
 	return nil
 }
 
+func getCalculations(awsConfig aws.Config, ec2Instances map[*model.ITResource]types.InstanceType) map[string]AWSCalculation {
```
Comment (leonardpahlke) [inline pkg/agent/targets/aws/target.go:103]: getCslculations is a too generic name IMO

Comment (leonardpahlke) [reply pkg/agent/targets/aws/target.go:103]: Also return an error

File: pkg/agent/targets/aws/target.go [hunk 2]
```diff
@@ -72,9 +89,79 @@ func (Target) Register(a any, r *prometheus.Registry) error {
...
+						Period: aws.Int32(300),
+						Stat:   aws.String("Average"),
+					},
+				},
+			},
+		}
+
+		result, err := GetMetrics(context.TODO(), cloudwatchClient, input)
+		if err != nil {
+			fmt.Println(err)
```
Comment (leonardpahlke) [inline pkg/agent/targets/aws/target.go:134]: Don't use panic. Return an error

File: pkg/agent/targets/aws/target.go [hunk 3]
```diff
@@ -72,9 +89,79 @@ func (Target) Register(a any, r *prometheus.Registry) error {
...
+		}
+
+		cpuUtilization := result.MetricDataResults[0].Values[0]
+		avgMinWatts := 0.74
+		avgMaxWatts := 3.5
+		avgDiffWatts := avgMaxWatts - avgMinWatts
```
Comment (leonardpahlke) [inline pkg/agent/targets/aws/target.go:140]: Define a config spec? cloudWatchMetricsConfig


Example:
'''go
type ConfigExample struct {
  Name string
  Type int
  // ...
}
func ExampleFunc(c *ConfigExample){}
'''

**Review by leonardpahlke — COMMENTED** [review]
@christopherfriedrich maybe you have some comments too.

**Comment by leonardpahlke** [comment]
@bcesur also have a look at the pipeline: https://app.circleci.com/pipelines/github/carbonaut-cloud/carbonaut/124/workflows/6e93c892-2ebc-4af8-a28d-aab9609f68f6

**Comment by christopherfriedrich** [comment]
Instead of having the watt usage as metric of the current form I would propose to do it this way:
```
aws_avg_watt_calculation{instance_id="i-007ecae81746e884a",provider="aws",service="ec2"} 5.9845879410954215
aws_computed_watt_hours_calculation{instance_id="i-007ecae81746e884a",provider="aws",service="ec2"} 143.63011058629013
```

File: pkg/agent/targets/aws/target.go [hunk 4]
```diff
@@ -72,9 +89,79 @@ func (Target) Register(a any, r *prometheus.Registry) error {
...
+
+		avgWatts := avgMinWatts + (cpuUtilization)*avgDiffWatts
+		computeWattHours := avgWatts * 24
+
+		emission := AWSCalculation{
+			InstanceId:        k.ServiceName,
+			AvgWatts:          fmt.Sprintf("%v", avgWatts),
```
[the remark this answers is not in the document]
Comment (bcesur) [reply pkg/agent/targets/aws/target.go:147]: it is a float. if you know a better way to parse float to string let me know 😄

File: pkg/agent/targets/aws/target.go (hunk 3 above)
Comment (bcesur) [reply pkg/agent/targets/aws/target.go:140]: I don't think we need a struct for these values here. I'll extract the calculations to another function.

**Review by christopherfriedrich — APPROVED** [review]

**Review by leonardpahlke — APPROVED** [review]

**Pull request merged**